app/controllers/api/v1/admin/domain_blocks_controller.rb

```diff
@@ -29,10 +29,13 @@ def show
   def create
     authorize :domain_block, :create?
 
+    @domain_block = DomainBlock.new(resource_params)
     existing_domain_block = resource_params[:domain].present? ? DomainBlock.rule_for(resource_params[:domain]) : nil
-    return render json: existing_domain_block, serializer: REST::Admin::ExistingDomainBlockErrorSerializer, status: 422 if existing_domain_block.present?
 
-    @domain_block = DomainBlock.create!(resource_params)
+    return render json: existing_domain_block, serializer: REST::Admin::ExistingDomainBlockErrorSerializer, status: 422 if existing_domain_block.present? && !@domain_block.stricter_than?(existing_domain_block)
+    return render json: existing_domain_block, serializer: REST::Admin::ExistingDomainBlockErrorSerializer, status: 422 if existing_domain_block.present? && existing_domain_block.domain == TagManager.instance.normalize_domain(@domain_block.domain)
```

Comment:
These two lines on top of each other that are basically the same is sort of weird... possible refactor ideas:

- Convert the full "is there an existing_domain_block present and is either the stricter check or the domain comparison true?" into at least one and maybe multiple private query methods to make this one more readable line?
- Extract `existing_domain_block` into a memoized private method instead of local var in this method?
- Convert the entire "existing check" into a before_action here, instead of in-line check in the action method?

Reply:
I don't really see the point in moving the local variable into a memoized private method.

Moving the check to a `before_action` could make sense, but we'd need to move the authorization check in such a filter as well, which is not the pattern we have adopted anywhere else.

I combined the two conditions into one, although I suppose I could further move that check to a private method.

Reply:
> I don't really see the point in moving the local variable into a memoized private method.

On it's own, probably not that useful. If you did extract a query method for the condition check and/or pull the whole thing out to a before_action, it would probably be useful/necessary in those scenarios.

> Moving the check to a `before_action` could make sense, but we'd need to move the authorization check in such a filter as well, which is not the pattern we have adopted anywhere else.

Ah, good point ... b/c we only want to even run this and do the existing if that first condition is passed. Given that, I'd keep this in the method and not bother with the before_action here.

> I combined the two conditions into one, although I suppose I could further move that check to a private method.

I personally think this would be more readable, but not a blocker.

Reply:
Updated to put the condition in a private method.